src/Kiota.Builder/Writers/CSharp/CodeIndexerWriter.cs

```diff
@@ -11,10 +11,21 @@ public override void WriteCodeElement(CodeIndexer codeElement, LanguageWriter wr
         if (codeElement.Parent is not CodeClass parentClass) throw new InvalidOperationException("The parent of a property should be a class");
         var returnType = conventions.GetTypeString(codeElement.ReturnType, codeElement);
         conventions.WriteShortDescription(codeElement.Documentation.Description, writer);
+
+        MarkObsoleteIfDeprecationInformationIsPresent(codeElement, writer);
+        
         writer.StartBlock($"public {returnType} this[{conventions.GetTypeString(codeElement.IndexType, codeElement)} position] {{ get {{");
         if (parentClass.GetPropertyOfKind(CodePropertyKind.PathParameters) is CodeProperty pathParametersProp)
             conventions.AddParametersAssignment(writer, pathParametersProp.Type, pathParametersProp.Name.ToFirstCharacterUpperCase(), string.Empty, (codeElement.IndexType, codeElement.SerializationName, "position"));
         conventions.AddRequestBuilderBody(parentClass, returnType, writer, conventions.TempDictionaryVarName, "return ");
         writer.CloseBlock("} }");
     }
+
+    private static void MarkObsoleteIfDeprecationInformationIsPresent(CodeIndexer codeElement, LanguageWriter writer)
+    {
+        if (codeElement is { DeprecationInformation.Reason: { } deprecationReason })
+            writer.WriteLine($"""[System.Obsolete("{deprecationReason.Replace("\"", "\\\"")}")]""");
```

Comment:
We can assume the string won't contain quotes. In what you are implementing it'll be a constant and for the other part we'll clean things up in the builder.

Reply:
Done.